p2p/src/network/yamux/p2p_network_yamux_reducer.rs

```diff
@@ -318,21 +349,45 @@ impl P2pNetworkYamuxState {
                             });
                         }
                     }
+                    YamuxFrameInner::WindowUpdate { .. } => {
+                        while let Some(frame) = pending_outgoing.pop_front() {
```

Comment:
Is there any bound or practical limit in how many elements we can have in `pending_outgoing`?
I'm concerned that someone could try to fill the dispatch queue and and cause blockings in the state machine